src/components/CommentCard.jsx

```diff
@@ -1,11 +1,57 @@
-export const CommentCard = ({author, created_at, body, votes}) => {
+import { useContext, useState, useEffect } from "react"
+import { toast } from "react-toastify";
+import "react-toastify/dist/ReactToastify.css";
+import { UserContext } from "../contexts/User";
+import { deleteComment } from "../utils/api";
+
+
+
+
+export const CommentCard = ({author, created_at, body, votes, comment_id, setComments}) => {
+    const userForCommentDelete = useContext(UserContext);
+    const [commentIdToDelete, setCommentIdToDelete] = useState(comment_id);
+    const [isClicked, setIsClicked] = useState(false);
+  
+
+    useEffect(() => {
+        if(isClicked) {
+           optimisticDeleteComment(commentIdToDelete)
+        deleteComment(commentIdToDelete).then(() => {
+
+        })
+        .catch((err) => {
+            handleError()   
+        })
+        }
+    }, [isClicked])
+
+
+    const handleError = () => {   
+        toast("There was a problem. Comment was not deleted.", {
+            position: "top-center",
+            bodyClassName: 'toastBody',
+        });
+    };
+    const handleDeleteClick = (event) => {
+        setIsClicked(true);
+     }
+    const optimisticDeleteComment = (commentIdToDelete) => {
+        console.log('first part');
+        setComments((currentComments) => 
+            currentComments.filter((comment) => comment.comment_id !== commentIdToDelete)
+        );
```

Comment:
I would be hesitant to delete _this_ optimistically. Deleting usually involves sensitive data and we wouldn't want a scenario where someone quickly deletes a controversial comment, sees it disappear then instantly logs off. The popup error message is great but we do have to _wait_ for the request to fail in order to see it. If we were using slow 3G this error message might be too late. 

There is no right/wrong for optimistic rendering these are just my thoughts. It's important you have reasoning for using it/not using it. 

I would consider placing optimisticDeleteComment() inside the .then block. That way you don't have to refresh to see it deleted, but you _do_ have to ensure the request has been fully successful first 👍